Sources/ResourceProvider/Cache/AnyAsyncCache.swift

```diff
@@ -0,0 +1,48 @@
+//
+//  AnyAsyncCache.swift
+//  swift-resource-provider
+//
+//  Created by Óscar Morales Vivó on 4/11/23.
+//
+
+/**
+ Type-erased async cache.
+
+ This wrapper value type can be (and is) used to build up adapters for actual cache types, and can also be used to
+ build mocks for testing.
+ */
+public struct AnyAsyncCache<ID: Hashable, Value> {
+    /**
+     A type erased cache has its functionality injected as blocks.
+     - Parameters:
+       - valueForID: Block that implements `AsyncCache.valueFor(id:)`
+       - storeValueForID: Block that implements `AsyncCache.store(value:id:)`
+     */
+    public init(
+        valueForID: @escaping (ID) async -> Value? = { _ in nil },
+        storeValueForID: @escaping (Value, ID) async -> Void = { _, _ in }
+    ) {
+        self.valueForID = valueForID
+        self.storeValueForID = storeValueForID
+    }
+
+    /// Implements `AsyncCache.valueFor(id:)`
```

Comment:
```suggestion
    /// Underlying implementation for ``AsyncCache/valueFor(id:)``.
```
Double backticks will get the doc links connected up.

Reply:
Ah forgot about those. Committing.